### PR TITLE
[jaeger-v2] add cassandra e2e integration tests

### PR DESCRIPTION
**Pull request by akagami-harsh**
## Which problem is this PR solving?
-  part of #5254 

## Description of the changes
- added cassandra integration tests
- added method to purge cassandra storage

## How was this change tested?
- some tests are failing

## Checklist
- [x] I have read https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md
- [x] I have signed all commits
- [x] I have added unit tests for the new functionality
- [x] I have run lint and test steps successfully
  - for `jaeger`: `make lint test`
  - for `jaeger-ui`: `yarn lint` and `yarn test`

### CODE REVIEW

File: .github/workflows/ci-cassandra.yml [hunk 1]
```diff
@@ -43,7 +44,7 @@ jobs:
         go-version: 1.22.x
 
     - name: Run cassandra integration tests
-      run: bash scripts/cassandra-integration-test.sh ${{ matrix.version.image }} ${{ matrix.version.schema }}
+      run: bash scripts/cassandra-integration-test.sh ${{ matrix.version.image }} ${{ matrix.version.schema }} ${{ matrix.jaeger-version }}
```
Comment (yurishkuro) [inline .github/workflows/ci-cassandra.yml:47]: add this to codecov flags below as well (L57)

File: scripts/cassandra-integration-test.sh [hunk 2]
```diff
@@ -52,11 +52,22 @@ apply_schema() {
...
   local schema_version=$2
+  local jaegerVersion=$3
   local cid
-  cid=$(setup_cassandra "${version}")
-  apply_schema "$2"
-  STORAGE=cassandra make storage-integration-test
-  exit_status=$?
+  if [ "${jaegerVersion}" = "v1" ]; then
+    cid=$(setup_cassandra "${version}")
+    apply_schema "$2"
```
Comment (yurishkuro) [inline scripts/cassandra-integration-test.sh:59]: these two lines are not different between two versions, you can do them outside of the if()

**Comment by akagami-harsh** [comment]
> most of the test are passing just these `FindTrace` tests are failing :
> 
> ```
> "Tags_in_one_spot_-_Tags",
> "Tags_in_one_spot_-_Logs",
> "Tags_in_one_spot_-_Process",
> "Tags_in_different_spots",
> ```

i tried some fixes but couldn't fix these test. do you have any clue why on the tests with `Tags....`  are failing

File: plugin/storage/cassandra/factory.go [hunk 3]
```diff
@@ -251,3 +252,7 @@ func (f *Factory) Close() error {
 func (f *Factory) PrimarySession() cassandra.Session {
 	return f.primarySession
 }
+
+func (f *Factory) Purge() error {
+	return f.primarySession.Query("TRUNCATE traces").Exec()
```
Comment (yurishkuro) [inline plugin/storage/cassandra/factory.go:257]: the v1 e2e test should also use this function

Comment (yurishkuro) [reply plugin/storage/cassandra/factory.go:257]: similar to ES tests, I would recommend first making this change to the existing integration test (i.e. switch it to use Purge and not recreate the factory for every test). Once that works supporting e2e test should be much simpler.

Comment (akagami-harsh) [reply plugin/storage/cassandra/factory.go:258]: #5414

File: cmd/jaeger/internal/integration/cassandra_test.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+			SkipList: []string{
+				"Tags_+_Operation_name_+_Duration_range",
+				"Tags_+_Duration_range",
+				"Tags_+_Operation_name_+_max_Duration",
+				"Tags_+_max_Duration",
+				"Operation_name_+_Duration_range",
+				"Duration_range",
+				"max_Duration",
+				"Multiple_Traces",
+			},
```
Comment (yurishkuro) [inline cmd/jaeger/internal/integration/cassandra_test.go:30]: I suggest defining this as a constant (CassandraSkippedTests) in v1 tests and reusing here.

**Comment by akagami-harsh** [comment]
the `FindTrace` tests are failing because the `vType` of `vBinary` tag is getting converted form `BINARY` to `STRING` . don't know yet which part of code is causing it. so should i modify the test comparison logic to handle this condition?

**Comment by yurishkuro** [comment]
We had the same issue with badger, and there is a flag in the main struct to skip binary tags

File: cmd/jaeger/internal/integration/cassandra_test.go (hunk 4 above)
Comment (akagami-harsh) [reply cmd/jaeger/internal/integration/cassandra_test.go]: added it in the `integration.go`

File: plugin/storage/cassandra/factory.go [hunk 5]
```diff
@@ -90,6 +91,9 @@ func NewFactoryWithConfig(
 	}
 	f := NewFactory()
 	f.primaryConfig = &cfg
+	f.Options.Index.Tags = true
+	f.Options.Index.Logs = true
+	f.Options.Index.ProcessTags = true
```
Comment (yurishkuro) [inline plugin/storage/cassandra/factory.go:96]: this doesn't make sense to me - the objective of this function is to create a factory with externally provided configuration. Instead, we're setting some configuration properties directly - why can they not be set in the caller?

**Review by yurishkuro — COMMENTED** [review]
lgtm aside from one question about config settings

File: plugin/storage/cassandra/factory.go (hunk 5 above)
Comment (akagami-harsh) [reply plugin/storage/cassandra/factory.go]: made index configurable and added above fields in config file

**Review by yurishkuro — APPROVED** [review]

**Comment by yurishkuro** [comment]
🎉 🎈

**Pull request merged**